src/api/spec/models/user_spec.rb

```diff
@@ -623,4 +623,48 @@
       end
     end
   end
+
+  describe '#can_modify?' do
+    let(:user) { create(:confirmed_user, :with_home, login: 'hans') }
+    let(:project) { user.home_project }
+
+    context 'when dealing with projects' do
+      context 'when checking against a home project' do
+        it { expect(user.can_modify?(project)).to be true }
+      end
+
+      context 'when checking against a child of a home project' do
+        let(:child_project) { create(:project, name: "#{project.name}:something") }
+
+        before do
+          child_project.relationships.build(user: user,
+                                            role: Role.find_by_title('maintainer'))
+        end
+
+        it { expect(user.can_modify?(child_project)).to be true }
+      end
+
+      context 'when the project has no parent project' do
```

Comment:
Not sure why you are testing this?

Reply:
Because before this change, `has_local_permission?` tries to go up the project hierarchy to check for permissions and stops when no parent is found, leaving you stranded without permissions.

Reply:
So? Who cares about before? :)

Reply:
Whoever changes that method and triggers a regression.

Reply:
Whoever changes that method will also change the spec...

Reply:
I'm fine with that, but at least it has to read the use case and think about what's going to happen. Don't you think?

Reply:
What use case? The use case that isn't considered anymore in the code? Test existing implementation and *not* the previous implementation or the knowledge of potential future committers of a previous implementation.

Reply:
Alright